packages/core/src/external/carequality/iti-55-parsing.ts

```diff
@@ -10,82 +10,101 @@ import { isAnyPatientMatching } from "./patient-matching";
-    ) {
-      livingSubjectId = {
-        extension: parameterList["livingSubjectId"][0]["value"][0]["$"]["extension"],
-        root: parameterList["livingSubjectId"][0]["value"][0]["$"]["root"],
-      };
-    }
-    let principalCareProviderId: PrincipalCareProviderId | undefined;
-    if (
-      parameterList["principalCareProviderId"] &&
-      parameterList["principalCareProviderId"][0]["value"] &&
-      parameterList["principalCareProviderId"][0]["value"][0]["$"]
-    ) {
-      principalCareProviderId = {
-        extension: parameterList["principalCareProviderId"][0]["value"][0]["$"]["extension"],
-        root: parameterList["principalCareProviderId"][0]["value"][0]["$"]["root"],
-      };
-    }
+  // these dont need to be defined
+  const phone =
+    result?.["Envelope"]?.["Body"]?.[0]?.["PRPA_IN201305UV02"]?.[0]?.["controlActProcess"]?.[0]?.[
+      "queryByParameter"
+    ]?.[0]?.["parameterList"]?.[0]?.["patientTelecom"]?.[0]?.["value"]?.[0]?.["$"]?.["value"];
+  const queryId =
+    result?.["Envelope"]?.["Body"]?.[0]?.["PRPA_IN201305UV02"]?.[0]?.["controlActProcess"]?.[0]?.[
+      "queryByParameter"
+    ]?.[0]?.["queryId"]?.[0]?.["$"]?.["extension"];
+  const root =
+    result?.["Envelope"]?.["Body"]?.[0]?.["PRPA_IN201305UV02"]?.[0]?.["id"]?.[0]?.["$"]?.["root"];
 
-    const patientAddress: Address = {
-      addressLine1: address["streetAddressLine"][0],
-      city: address["city"][0],
-      state: address["state"][0],
-      zip: address["postalCode"][0],
-      country: address["country"] ? address["country"][0] : "USA",
-    };
+  const livingSubjectId: LivingSubjectId | undefined = parameterList["livingSubjectId"]?.[0]?.[
+    "value"
+  ]?.[0]?.["$"]
+    ? {
+        extension: parameterList["livingSubjectId"][0]["value"][0]["$"]?.["extension"],
+        root: parameterList["livingSubjectId"][0]["value"][0]["$"]?.["root"],
+      }
+    : undefined;
 
-    const patientData = {
-      livingSubjectId: livingSubjectId,
-      principalCareProviderId: principalCareProviderId,
-      firstName: patientName["given"][0],
-      lastName: patientName["family"][0],
-      dob: dob,
-      genderAtBirth: gender,
-      address: [patientAddress],
-      contact: [{ phone: phone }],
-    };
-    return [patientData, [root, extension, signature]];
-  } catch (error) {
-    console.error(error);
-    throw new Error("XML parsing failed");
-  }
+  const principalCareProviderId: PrincipalCareProviderId | undefined = parameterList[
+    "principalCareProviderId"
+  ]?.[0]?.["value"]?.[0]?.["$"]
+    ? {
+        extension: parameterList["principalCareProviderId"][0]["value"][0]["$"]?.["extension"],
+        root: parameterList["principalCareProviderId"][0]["value"][0]["$"]?.["root"],
+      }
+    : undefined;
+
+  const patientData = {
+    livingSubjectId: livingSubjectId,
+    principalCareProviderId: principalCareProviderId,
+    firstName: patientName["given"][0],
+    lastName: patientName["family"][0],
```

Comment:
This can throw runtime error. Is this expected to be thrown by the function w/o any handling?